server/utils/visitorUtils.ts

```diff
@@ -53,7 +53,7 @@ export const getBanStatus = (restrictions: Restriction[]): BanStatus => {
     return { isBanned: false }
   }
   // if there is a ban with no end date no further checks needed
-  if (banned.find(ban => ban.expiryDate === undefined)) {
+  if (banned.find(ban => !ban.expiryDate)) {
```

Comment:
As mentioned above, depending on the source, `expiryDate` could be `null` or `undefined`